### PR TITLE
Export public types, update DU config

### PR DESCRIPTION
**Pull request by teogeb**
Export parameters/return types for objects which are publicly available to the JS-client users. 

Updated `getDataUnionMainnetAddress` and `getDataUnionSidechainAddress` in `Contracts.ts` to support new DataUnion addresses.

BREAKING CHANGE: 
- moved DataUnion related StreamrClient options into a new `dataUnion` block: `factoryMainnetAddress`, `factorySidechainAddress`, `minimumWithdrawTokenWei` 
- renamed `payForSignatureTransport` to `freeWithdraw` and reversed the value
- new required options in that block: `templateMainnetAddress`, `templateSidechainAddress`

### CODE REVIEW

File: src/dataunion/Contracts.ts [hunk 1]
```diff
@@ -37,11 +41,15 @@ export class Contracts {
...
 
     getDataUnionMainnetAddress(dataUnionName: string, deployerAddress: EthereumAddress) {
-        if (!this.factoryMainnetAddress) {
-            throw new Error('StreamrClient has no factoryMainnetAddress configuration.')
+        if (!isAddress(this.factoryMainnetAddress)) {
+            throw new Error('StreamrClient factoryMainnetAddress configuration is ' + this.factoryMainnetAddress ? 'missing' : 'not a valid Ethereum address')
+        }
+
+        if (!isAddress(this.templateMainnetAddress)) {
+            throw new Error('StreamrClient dataUnionTemplateMainnetAddress configuration is ' + this.templateMainnetAddress ? 'missing' : 'not a valid Ethereum address')
```
Comment (timoxley) [inline src/dataunion/Contracts.ts:49]: I think this ternary is around the wrong way, 

```suggestion
            throw new Error('StreamrClient dataUnionTemplateMainnetAddress configuration is ' + this.templateMainnetAddress ? 'not a valid Ethereum address' : 'missing')
```

File: src/dataunion/Contracts.ts [hunk 2]
```diff
@@ -53,11 +61,15 @@ export class Contracts {
     }
 
     getDataUnionSidechainAddress(mainnetAddress: EthereumAddress) {
-        if (!this.factorySidechainAddress) {
-            throw new Error('StreamrClient has no factorySidechainAddress configuration.')
+        if (!isAddress(this.factorySidechainAddress)) {
+            throw new Error('StreamrClient factorySidechainAddress configuration is ' + this.factorySidechainAddress ? 'missing' : 'not a valid Ethereum address')
```
Comment (timoxley) [inline src/dataunion/Contracts.ts:65]: I think this ternary is around the wrong way, 

```suggestion
            throw new Error('StreamrClient factorySidechainAddress configuration is ' + this.factorySidechainAddress ? 'not a valid Ethereum address' : 'missing')
```

File: src/dataunion/Contracts.ts [hunk 3]
```diff
@@ -53,11 +61,15 @@ export class Contracts {
...
         }
-        // NOTE! this must be updated when DU sidechain smartcontract changes: keccak256(CloneLib.cloneBytecode(data_union_sidechain_template))
-        const codeHash = '0x040cf686e25c97f74a23a4bf01c29dd77e260c4b694f5611017ce9713f58de83'
+
+        if (!isAddress(this.templateSidechainAddress)) {
+            throw new Error('StreamrClient dataUnionTemplateSidechainAddress configuration is ' + this.templateSidechainAddress ? 'missing' : 'not a valid Ethereum address')
```
Comment (timoxley) [inline src/dataunion/Contracts.ts:69]: I think this ternary is around the wrong way, 

```suggestion
            throw new Error('StreamrClient dataUnionTemplateSidechainAddress configuration is ' + this.templateSidechainAddress ? 'not a valid Ethereum address' : 'missing')
```

**Review by timoxley — APPROVED** [review]
@teogeb @jtakalai can you verify those invalid address error messages are correct? but otherwise lgtm. 👍

File: src/dataunion/Contracts.ts [hunk 4]
```diff
@@ -37,11 +41,15 @@ export class Contracts {
     }
 
     getDataUnionMainnetAddress(dataUnionName: string, deployerAddress: EthereumAddress) {
-        if (!this.factoryMainnetAddress) {
-            throw new Error('StreamrClient has no factoryMainnetAddress configuration.')
+        if (!isAddress(this.factoryMainnetAddress)) {
+            throw new Error('StreamrClient factoryMainnetAddress configuration is ' + this.factoryMainnetAddress ? 'missing' : 'not a valid Ethereum address')
```
[the remark this answers is not in the document]
Comment (teogeb) [reply src/dataunion/Contracts.ts]: Fixed the ternaries

**Pull request merged**